test/wdio/slot-array-complex/cmp.test.tsx

```diff
@@ -0,0 +1,21 @@
+import { h } from '@stencil/core';
+import { render } from '@wdio/browser-runner/stencil';
+
+describe('slot array complex', () => {
+  beforeEach(async () => {
+    render({
+      template: () => <slot-array-complex-root></slot-array-complex-root>,
+    });
+
+    await $('main slot-array-complex').waitForExist();
+  });
+
+  it('renders slotted content', async () => {
+    await expect($('main slot-array-complex')).toHaveChildren(3);
+    await expect($('main slot-array-complex').$$('*')[0]).toHaveText('slot - start');
+    await expect($('main slot-array-complex').$$('*')[1]).toHaveText('slot - default');
+    await expect($('main slot-array-complex').$$('*')[2]).toHaveText('slot - end');
```

Comment:
This will be a bit more effective:
```suggestion
    const elems = $('main slot-array-complex').$$('*')
    await expect(elems[0]).toHaveText('slot - start');
    await expect(elems[1]).toHaveText('slot - default');
    await expect(elems[2]).toHaveText('slot - end');
```